### ISSUE TITLE
Endless loop on mousemove

### ISSUE THREAD

**Issue by mpanin** [gemini-testing/gemini#651]
Output of `gemini --version`:

3.10.8

Contents of `.gemini.yml` file:

rootUrl: http://our_website.html
gridUrl: http://hub_locatio/wd/hub
compositeImage: true
screenshotsDir: 'screens/mobile'
httpTimeout: 180000
sessionRequestTimeout: 180000
retry: 5

system:
  debug: false

browsers:
  chrome-latest:
    desiredCapabilities:
      browserName: chrome
      screenshotMode: fullpage
      chromeOptions:
        mobileEmulation: {
          deviceName: 'Apple iPhone 6'
        }

Output from selenium node:
20:54:21.241 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return typeof __gemini !== "undefined"? __gemini.prepareScreenshot([".Title.sponsored"], {"coverage":false,"ignoreSelectors":[],"usePixelRatio":true}) : {error: "ERRNOFUNC"};, []])
20:54:21.328 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return typeof __gemini !== "undefined"? __gemini.prepareScreenshot([".Title.sponsored"], {"coverage":false,"ignoreSelectors":[],"usePixelRatio":true}) : {error: "ERRNOFUNC"};, []]
20:54:22.122 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [take screenshot])
20:54:22.652 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [take screenshot]
20:54:22.867 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return document.body;, []])
20:54:22.867 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return document.body;, []]
20:54:22.898 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [mousemove: 2 true])
20:54:22.952 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [mousemove: 2 true]
20:54:22.979 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return document.body;, []])
20:54:22.984 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return document.body;, []]
20:54:23.008 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [mousemove: 2 true])
20:54:23.012 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [mousemove: 2 true]
20:54:23.085 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return document.body;, []])
20:54:23.101 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return document.body;, []]
20:54:23.163 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [mousemove: 2 true])
20:54:23.163 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [mousemove: 2 true]
20:54:23.195 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return document.body;, []])
20:54:23.195 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return document.body;, []]
20:54:23.229 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [mousemove: 2 true])
20:54:23.236 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [mousemove: 2 true]
20:54:23.257 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return document.body;, []])
20:54:23.257 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return document.body;, []]
20:54:23.288 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [mousemove: 2 true])
20:54:23.288 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [mousemove: 2 true]
20:54:23.320 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return document.body;, []])
20:54:23.327 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return document.body;, []]
20:54:23.350 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [mousemove: 2 true])
20:54:23.350 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [mousemove: 2 true]
20:54:23.398 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return document.body;, []])
20:54:23.398 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return document.body;, []]
20:54:23.430 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [mousemove: 2 true])
20:54:23.437 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [mousemove: 2 true]
20:54:23.460 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [execute script: return document.body;, []])
20:54:23.465 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [execute script: return document.body;, []]
20:54:23.488 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Executing: [mousemove: 2 true])
20:54:23.495 INFO [15] org.openqa.selenium.remote.server.DriverServlet - Done: [mousemove: 2 true]
...

Result:

I've noticed that it happens only on latest version of Gemini, I've updated from 4.9.4 and never seen issue on that version. It seems that it happens only during retries.

Expected behavior:

Tests are successfully re-executed.

**Comment by vincefliek** [comment]
@mpanin I have the same trouble
Agree with you, this strange behaviour is on the latest version, I've updated today
my temporary solution was to disable retries in .gemini.yml

**Comment by mpanin** [comment]
@vincefliek 
Unfortunately disabling retries is not an option for us, they are pretty helpful in several cases.

Also, I've noticed that this issues is actual on older versions as well, however, it seems that on older versions all retries just fail and capture only top of the page.

**Comment by sipayRT** [comment]
hi. can you provide your test on which it happens?

**Comment by mpanin** [comment]
Well, we have a little wrapper over Gemini tests, but after compiling test looks like this:
YML config is provided above, site URL is: www.thedailybeast.com

```
   global.gemini.suite(testBaseName, (suite) => {
      suite
        .setUrl(`/articles/2016/07/11/pokemon-go-is-a-hacker-s-dream.html`)
        .before((actions) => {
        });

        suite
          .setCaptureElements('.ArticleBody .Text:nth-of-type(n+4)')
          .capture('root', { tolerance: tolerance }, (actions) => {
            actions.waitForElementToShow('.ArticleBody .Text:nth-of-type(n+4)', 5000);
          });
    });
  }
}
```

Actually I think that I was able to find work around this issue by setting` suitesPerSession: 1`, at least I'm no longer able to this issue appearing (all retries have different screens), so probably it's some how related to the fact that we're retrying tests in the same browser instance.

**Comment by sipayRT** [comment]
> we're retrying tests in the same browser instance

no, each retry will be executed in new browser session. So problem can be in another place. I'll try to reproduce your problem

**Comment by sipayRT** [comment]
@mpanin are you sure that you need to take screenshot of 76 elements?
```js
document.querySelectorAll('.ArticleBody .Text:nth-of-type(n+4)').length
76
```

**Comment by mpanin** [comment]
@sipayRT seems like we're capturing only first one that matches, since in results I see only first element that matches selector, probably smthn like `.ArticleBody .Text:nth-of-type(5)` is going to be a better solution.

**Comment by sipayRT** [comment]
can you reproduce it with `calibrate: false` option?

**Comment by mpanin** [comment]
Having problem reproducing this issue with both `calibrate: false `and `calibrate: true`, makes sense to put this issue on hold.
Maybe @vincefliek is able to help.

**Comment by leonsabr** [comment]
@mpanin please check if this issue is fixed in release v4.18.2.

**Comment by mpanin** [comment]
@leonsabr Unfortunately I don't have access to this Gemini project anymore, so case no one else in here is able to verify this issue - feel free to close it without my confirmation.
